### PR TITLE
alternate_protocols_cache: Move alternate_protocols_cache logging to the alternate_protocols_cache log ID.

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
alternate_protocols_cache: Move alternate_protocols_cache logging to the alternate_protocols_cache log ID.

Also add a new doc which describes how Envoy logging works.

Risk Level: Low
Testing: N/A
Docs Changes: Add a new doc: source/docs/logging.md
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
@RenjieTang hopefully this matches what we discussed offline today and hopefully logging.md explains how this works. I would greatly appreciate any suggestions you might have for how to improve the docs!

File: source/docs/logging.md
Comment (alyssawilk) [inline source/docs/logging.md:71]: s/TO_//

File: source/docs/logging.md
Comment (alyssawilk) [inline source/docs/logging.md:86]: optional nit I'd be inclined to put in nonsense numbers so C123 S345 but thtally your call

**Review by alyssawilk — COMMENTED** [review]
Thanks for catching the ID snafu and hugely improving our docs in this space.
What do you think of adding this md file to the logger.h file around where the do_not_use macro is, so folks who look it up are more likely to find the explanation?

File: source/extensions/filters/http/alternate_protocols_cache/filter.h
Comment (alyssawilk) [inline source/extensions/filters/http/alternate_protocols_cache/filter.h:43]: I'm not entirely clear if we should have a cache_filter log and a cache log, but as the only use case for this is logging something is about to go in the cache I'm inclined to leave it the same ID for now unless another maintainer objects :-)

File: source/docs/logging.md
Comment (RyanTheOptimist) [reply source/docs/logging.md]: Doh! Thanks, done!

File: source/docs/logging.md
Comment (RyanTheOptimist) [reply source/docs/logging.md]: Sold! Done.

File: source/extensions/filters/http/alternate_protocols_cache/filter.h
Comment (RyanTheOptimist) [reply source/extensions/filters/http/alternate_protocols_cache/filter.h:43]: SGTM :)

**Comment by RyanTheOptimist** [comment]
> Thanks for catching the ID snafu and hugely improving our docs in this space.
> What do you think of adding this md file to the logger.h file around where the do_not_use macro is, so folks who look it up are more likely to find the explanation?

Oh, shoot! I meant to do that but forgot. Thanks for the reminder. Done!

**Review by alyssawilk — DISMISSED** [review]
LGTM though I'm inclined to wait for Renjie's sign off too since I'd like a second read on if the docs are clear, especially to non-envoy-devs

**Comment by RyanTheOptimist** [comment]
> LGTM though I'm inclined to wait for Renjie's sign off too since I'd like a second read on if the docs are clear, especially to non-envoy-devs

Agreed! We discussed this offline and I’d be curious how this doc reads to him.

**Review by RenjieTang — APPROVED** [review]
Looks very clear to me. Thanks for doing this!

**Comment by alyssawilk** [comment]
Happy to submit once CI is happy
/wait until then

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**
